process/logs/logs_processor.go

```diff
@@ -1,18 +1,76 @@
 package logs
 
 import (
+	"github.com/ElrondNetwork/covalent-indexer-go"
+	"github.com/ElrondNetwork/covalent-indexer-go/process/utility"
 	"github.com/ElrondNetwork/covalent-indexer-go/schema"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
 	"github.com/ElrondNetwork/elrond-go-core/data"
 )
 
-type logsProcessor struct{}
+type logsProcessor struct {
+	pubKeyConverter core.PubkeyConverter
+}
 
 // NewLogsProcessor creates a new instance of logs processor
-func NewLogsProcessor() (*logsProcessor, error) {
-	return &logsProcessor{}, nil
+func NewLogsProcessor(pubKeyConverter core.PubkeyConverter) (*logsProcessor, error) {
+	if check.IfNil(pubKeyConverter) {
+		return nil, covalent.ErrNilPubKeyConverter
+	}
+
+	return &logsProcessor{pubKeyConverter: pubKeyConverter}, nil
 }
 
 // ProcessLogs converts logs data to a specific structure defined by avro schema
-func (lp *logsProcessor) ProcessLogs(logs map[string]data.LogHandler) ([]*schema.Log, error) {
-	return nil, nil
+func (lp *logsProcessor) ProcessLogs(logs map[string]data.LogHandler) []*schema.Log {
+	allLogs := make([]*schema.Log, 0)
```

Comment:
allLogs := make([]*schema.Log, 0, len(logs))

Reply:
this can be renamed. don't like the current name. maybe you can rename in `dbLogs` or something that suggests that you have converted the logs.